service/raft/group0_fwd.hh

```diff
@@ -8,10 +8,48 @@
 
 #pragma once
 
-#include <iostream>
+#include <iosfwd>
+#include "raft/raft.hh"
+#include "gms/inet_address.hh"
 
 namespace service {
 
+// Address of a discovery peer
+struct discovery_peer {
+    raft::server_id id;
+    gms::inet_address ip_addr;
+    bool operator==(const discovery_peer& rhs) const {
+        return ip_addr == rhs.ip_addr;
+    }
+
+    bool operator<(const discovery_peer& rhs) const {
```

Comment:
`operator<` should be consistent with `operator==`, which compares using `ip_addr`.

Reply:
Is `operator<` even needed for anything?

Reply:
BTW. you could implement `operator<=>` instead if you need those comparisons:
https://en.cppreference.com/w/cpp/language/default_comparisons

Reply:
Yes it is. No it should not. It's used in std::sort to find the smallest peer.